PowerShellTools.HostService/ServiceManagement/Debugging/PowershellDebuggingServiceUtilities.cs

```diff
@@ -286,5 +286,35 @@ private void objects_DataAdded(object sender, DataAddedEventArgs e)
                 NotifyOutputString(outputString.ToString());
             }
         }
+
+        /// <summary>
+        /// Adds a breakpoint record to the _psBreakpointTable while first checking that an bp with the same id 
+        /// does not already exist in the table.
+        /// </summary>
+        /// <param name="bpRecord"></param>
+        private void addToBpTable(PowerShellBreakpointRecord bpRecord)
+        {
+            if (bpTableDoesNotContain(bpRecord.Id))
+            {
+                _psBreakpointTable.Add(bpRecord);

```

Comment:
I asked myself this very same question when I realized we left ourselves the ability to add duplicates (and yes, a user really can make this happen) by using a List. I definitely think changing to a more appropriate data structure (set, hashset, etc.) would be good. Anyone else have thoughts on this? @EricMSFT  @AndreSayreMSFT  @HuanhuanSunMSFT  @adamdriscoll